### ISSUE TITLE
feat : putIfAbsent , lazyPutIfAbsent ,  factoryIfAbsent

### ISSUE THREAD

**Issue by maxzod** [maxzod/queen#7]

**Comment by hatemragab** [comment]
Why this we should make the following
If you tried to put existing instance it will return it with out any throw
And if not exist just put it 🙂

**Comment by maxzod** [comment]
to limit Magic little bit  😃
and to make sure the registered objects is the one i have added i the cases i want to make sure 
and the cases i just need it to be in the container no matter is mine or not 

for example Shared Pref instance it just need to make sure its already in the container so i will use `putIfAbsent`

and for other Service implementation  i need to make sure my implementation is the one used in the conatiner so i will use `put`  (if already there it will give me exception )

**Comment by hatemragab** [comment]
Disagree
Why we need to create put and putIfAbsent
We just need to simplify everything
Just put will do all you want we don't need make the screen red in the UI!

**Comment by hatemragab** [comment]
```
 static I _insert<I>(
    DependencyManager<I> manager, {
    String? tag,
    bool force = false,
  }) {
    final obj = _get<I>(tag);
    if (obj != null) return obj;
    final key = _buildKey<I>(tag);
    container[key] = manager;
    return _get<I>(tag)!;
  }
```
there is the new implemention of new insert method

**Comment by hatemragab** [comment]
i will delete 
```  bool force = false```
parameter

**Comment by maxzod** [comment]
fixed on [4d40e594daccc3de92c4846fb99015ed240e4688](https://github.com/maxzod/queen/commit/4d40e594daccc3de92c4846fb99015ed240e4688)
